docs/formatter.md

```diff
@@ -0,0 +1,648 @@
+behavior.
+
+When using Ruff as a formatter, we recommend disabling the following rules:
+
+- [`line-too-long`](rules/line-too-long.md)
+- [`single-line-implicit-string-concatenation`](rules/single-line-implicit-string-concatenation.md)
+- [`missing-trailing-comma`](rules/missing-trailing-comma.md)
+- [`prohibited-trailing-comma`](rules/prohibited-trailing-comma.md)
+- [`bad-quotes-inline-string`](rules/bad-quotes-inline-string.md)
+- [`bad-quotes-multiline-string`](rules/bad-quotes-multiline-string.md)
+- [`bad-quotes-docstring`](rules/bad-quotes-docstring.md)
+- [`avoidable-escaped-quote`](rules/avoidable-escaped-quote.md)
+
+Similarly, we recommend disabling the following isort settings, which are incompatible with the
+formatter's treatment of import statements when set to non-default values:
+
+- [`lines-after-imports`](https://docs.astral.sh/ruff/settings/#isort-lines-after-imports)
+- [`lines-between-types`](https://docs.astral.sh/ruff/settings/#isort-lines-between-types)
+
+## Exit codes
+
+`ruff format` exits with the following status codes:
+
+- `0` if no files were formatted.
+- `1` if any files were formatted.
+- `2` if Ruff terminates abnormally due to invalid configuration, invalid CLI options, or an
+    internal error.
+
+`ruff format --check` uses the same exit codes, such that an exit code of `1` indicates that a
+file _would_ be formatted if `--check` were not specified, while an exit code of `0` indicates that
+no files would be formatted.
+
+## Black compatibility
+
+The formatter is designed to be a drop-in replacement for [Black](https://github.com/psf/black).
+
+Specifically, the formatter is intended to emit near-identical output when run over Black-formatted
+code. When run over extensive Black-formatted projects like Django and Zulip, > 99.9% of lines
+are formatted identically. When migrating an existing project from Black to Ruff, you should expect
+to see a few differences on the margins, but the vast majority of your code should be unchanged.
+
+When run over _non_-Black-formatted code, the formatter makes some different decisions than Black,
+and so more deviations should be expected, especially around the treatment of end-of-line comments.
+
+If you identify deviations in your project, spot-check them against the [intentional deviations](#intentional-deviations)
+enumerated below, as well as the [unintentional deviations](https://github.com/astral-sh/ruff/issues?q=is%3Aopen+is%3Aissue+label%3Aformatter)
+filed in the issue tracker. If you've identified a new deviation, please [file an issue](https://github.com/astral-sh/ruff/issues/new).
+
+### Preview style
+
+Black gates formatting changes behind a [`preview`](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#preview-style)
+flag. The formatter does not yet support Black's preview style, though the intention is to support
+it within the coming months.
```

Comment:
We should add some words about how the black preview style will become the black stable style yearly and will do the same, even though not necessarily on the same timeline (right? i don't think we really talked about timelines for this, either way i think this should be done with a minor version bump)